src/commands/variables/update.ts

```diff
@@ -21,23 +23,50 @@ export default class UpdateVariable extends UpdateCommand<CreateVariableParams>
         featurePrompt
     ]
 
-    public async run(): Promise<void> {
-        await this.requireProject()
-        const { variable } = await inquirer.prompt([variablePrompt], {
-            token: this.authToken,
-            projectKey: this.projectKey
-        })
+    static flags = {
+        ...UpdateCommand.flags,
+        'description': Flags.string({
+            description: 'Description for the variable',
+        }),
+        'feature': Flags.string({
+            description: 'The ID of the feature to associate the variable to'
+        }),
+    }
 
-        this.writer.blankLine()
-        this.writer.statusMessage('Current values:')
-        this.writer.statusMessage(JSON.stringify(variable, null, 2))
-        this.writer.blankLine()
+    public async run(): Promise<void> {
+        const { args, flags } = await this.parse(UpdateVariable)
+        const { key } = args
+        const { headless } = flags
 
-        const params = await this.populateParameters(CreateVariableParams)
+        await this.requireProject()
+        let variableKey: string
+        if (headless) {
+            if (!key) {
+                this.writer.showError('The key argument is required')
+                return
+            }
+            variableKey = key
+        } else {
+            const { variable } = await inquirer.prompt<VariablePromptResult>([variablePrompt], {
```

Comment:
what happens when I put in a variable key that doesn't exist, won't this return nothing?

Reply:
well in non headless mode you can't put in a variable key that doesn't exist because you are prompted to select from a set of keys in your project. But if you do headless mode and you put a key that doesn't exist, then we show an error that looks like `DevCycle Error: Variable with key {key} not found`. We're just logging the api response.